packages/features/bookings/lib/getBookingResponsesSchema.ts

```diff
@@ -98,12 +98,13 @@ function preprocess<T extends z.ZodType>({
         const views = bookingField.views;
         const isFieldApplicableToCurrentView =
           currentView === "ALL_VIEWS" ? true : views ? views.find((view) => view.id === currentView) : true;
+        let hidden = bookingField.hidden;
+        const numOptions = bookingField.options?.length ?? 0;
+        if (bookingField.hideWhenJustOneOption) {
+          hidden = hidden || numOptions <= 1;
+        }
```

Comment:
```suggestion
        const numOptions = bookingField.options?.length ?? 0;
        const isRequired = !bookingField.hidden && isFieldApplicableToCurrentView && bookingField.required;
        if (bookingField.hideWhenJustOneOption && numOptions <= 1) {
          isRequired = false;
        }
```

Reply:
I find the current approach better to understand logically.
Whether a field is hidden or not is decided either by `bookingField.hidden`(which is explicitly decided by the user) or decided automatically by Cal if that field has one option(assuming hideWhenJustOneOption is true)

Now, whether a field is required or not is decided by 3 factors(
- It shouldn't be hidden
- It shouldn't be non-applicable to the current view
- It should be required.